core/src/Revolution/Sources/modMediaSource.php

```diff
@@ -1148,7 +1148,15 @@ public function uploadObjectsToContainer($container, array $objects = [])
             }
 
             if ((bool)$this->xpdo->getOption('upload_translit')) {
-                $newName = $this->xpdo->filterPathSegment($file['name']);
+                $uploadRegex = $this->xpdo->getOption('upload_translit_restrict_chars_pattern');
+                $options = !empty($uploadRegex)
+                    ? [
+                        'friendly_alias_restrict_chars' => 'pattern',
+                        'friendly_alias_restrict_chars_pattern' => $uploadRegex
+                    ]
+                    : []
+                    ;
+                $newName = $this->xpdo->filterPathSegment($file['name'], $options);
```

Comment:
I wonder if we shouldn't have a dedicated modMediaSource method instead of using filterPathSegment()…

Reply:
I don't _think_ so, as we'd end up with a lot of duplicated logic, especially taking the translit functionality into account. What I can see is moving the meat of the `filterPathSegment` fn out of `modResource` and into the one in the base `modX` class; I'm actually curious to know why it's wired the way it currently is (with `filterPathSegment` residing in both classes). 

Looking ahead, perhaps fns like this should be moved into their own (perhaps static) utility class(es)?

Reply:
Let's keep future discussion out of these comments. These methods were intended for dealing with Resource aliases only. They were never meant for file management. The modX helper method was just that—a convenience for calling the static method on the modResource class. I'm not sure why it was reused in media sources.

Reply:
Just ignore my original comment and we'll do it like this.